redisinsight/ui/src/components/global-url-handler/GlobalUrlHandler.tsx

```diff
@@ -29,8 +32,18 @@ const GlobalUrlHandler = () => {
   const history = useHistory()
   const dispatch = useDispatch()
   const location = useLocation()
+  const searchParams = new URLSearchParams(location.search)
+  const returnUrl = searchParams.get('returnUrl')
 
   useEffect(() => {
+    if (returnUrl) {
+      localStorageService.set(AppStorageItem.returnUrl, { value: returnUrl })
+      dispatch(setReturnUrl(returnUrl))
+      history.push(location.pathname)
+    } else if (localStorageService.get(AppStorageItem.returnUrl)) {
+      dispatch(setReturnUrl(localStorageService.get(AppStorageItem.returnUrl).value))
```

Comment:
move this logic with localStorageService.get to appUrlHandlingSlice
like here
https://github.com/RedisInsight/RedisInsight/blob/1e993142a3d3707e92f84d17d916553fa4fb3c72/redisinsight/ui/src/slices/user/user-settings.ts#L18